### PR TITLE
Added FieldWidgetThirdPartySettingsFormEvent.

### PR DESCRIPTION
**Pull request by Ambient-Impact**
See #103. Will merge duplicate code in follow-up PR.

### CODE REVIEW

File: modules/field_event_dispatcher/field_event_dispatcher.services.yml [hunk 1]
```diff
@@ -3,3 +3,8 @@ services:
     class: Drupal\field_event_dispatcher\EventSubscriber\Form\FormEntityViewDisplayEditAlterSubscriber
     tags:
       - { name: 'event_subscriber' }
+field_event_dispatcher.form_entity_form_display_edit_alter_event_subscriber:
```
Comment (pdenooijer) [inline modules/field_event_dispatcher/field_event_dispatcher.services.yml]: Please add a new line between service defs.
Add missing 2 spaces before this.
Update the naming with the above naming as an example.

File: modules/field_event_dispatcher/field_event_dispatcher.services.yml [hunk 2]
```diff
@@ -3,3 +3,8 @@ services:
...
+    class: \Drupal\field_event_dispatcher\EventSubscriber\Form\FormEntityFormDisplayEditAlterEventSubscriber
+    arguments: []
```
Comment (pdenooijer) [inline modules/field_event_dispatcher/field_event_dispatcher.services.yml]: This is not used so please remove it.

File: modules/field_event_dispatcher/src/EventSubscriber/Form/FormEntityFormDisplayEditAlterEventSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    $form = &$event->getForm();
+
+    // $form['#fields'] lists all field name keys on this entity.
+    foreach ($form['#fields'] as $fieldName) {
+      // Skip any fields that have no third-party settings.
+      if (!isset($form['fields'][$fieldName]['plugin']['settings_edit_form']['third_party_settings']['field_event_dispatcher'])) {
+        continue;
+      }
+
+      $fieldSettings = &$form['fields'][$fieldName]['plugin']['settings_edit_form'];
```
Comment (pdenooijer) [inline modules/field_event_dispatcher/src/EventSubscriber/Form/FormEntityFormDisplayEditAlterEventSubscriber.php]: Please use the third part settings instead so you don't need the key anymore 🙂:
`$thirdPartySettings = &$form['fields'][$fieldName]['plugin']['settings_edit_form']['third_party_settings'];`

**Review by pdenooijer — CHANGES_REQUESTED** [review]

**Comment by Ambient-Impact** [comment]
That's what I get for opening a PR right before I go to bed and not thoroughly checking the rebase beforehand. 🤓

**Review by pdenooijer — APPROVED** [review]

**Pull request merged**